### DISCUSSION TITLE
Smooth cut boundaries with hex mesh

### DISCUSSION THREAD

**Discussion by danishTUE** [neperfepx/neper#445] [Q&A] [answered]
Hi Romain,

I am interested to know if its possible in neper to get smooth boundaries with 'hex' element -type in the mesh of an RVE which is transformed by a cut? I am interested to get a hex mesh of a 3D RVE which is cut at the surface by a cylinder. Due to the cut (and thus non-planar surface), I end up having a jagged (stepped) kind of mesh at the surface of the cut, making the surface of the cut non-smooth (which can act as stress-concentrations in my analysis). However, for the same geometry I am able to generate generate all 'hex' mesh of the same geometry in Abaqus with smooth surface of the cut. I am interested to know if its possible to with neper as well?
Thanks in advance!

Regards,

Danish

**Comment by rquey** [comment, marked as answer]
See #98 for a possibility.

**Reply by danishTUE** [reply]
Thanks. I tried the work around example given by you in the previous thread. External manipulation of the mesh using gmsh seems to run smoothly and I generated a msh2 format file with it for the example given by you. However, when I try to run Visualization module over it, neper seems to crash with segmentati
[Quad.zip](https://github.com/rquey/neper/files/8516890/Quad.zip)
![Error](https://user-images.githubusercontent.com/95438134/164113897-231d8a76-8ecc-4d47-ae3b-86af7880b045.PNG)
on fault. I am attaching the Quad.msh file (which I generated by following the example given by you in the thread #98 ). Can you please take a look and let me know what may be the issue? Thanks in advance.

**Reply by rquey** [reply]
Add `-format msh22` to the `gmsh -2 *.geo` command to get Gmsh to write at the version 2.2 format, which Neper is able to read.

**Reply by Laowulve** [reply]
Hi, I'm curious as to how Neper is used to divide the surface cut by the cylinder, as I'm trying this now. Because of the direct use of 'hex' elements, the curved segments at the edges would be very discontinuous and would have an impact on the stress analysis. Thank you very much :)
![微信截图_20230403172530](https://user-images.githubusercontent.com/44744321/229469505-a2ad9c56-f00d-43fe-92c2-0fe0000c1ba1.png)

**Reply by rquey** [reply]
This is the best you can get from Neper's hex meshing.

To get a smooth surface, mesh in tets and try to convert to a hex mesh using this: https://github.com/neperfepx/neper/discussions/98.

**Reply by Laowulve** [reply]
Thanks for the reply. I am now trying to use Neper-generated seed points to classify crystal sets of elements meshed by ABAQUS. And I want to know how the seed_weights apply to the distance?
![image](https://user-images.githubusercontent.com/44744321/230329036-84eb43ae-ff60-4262-8181-37dad6f50c51.png)

**Reply by rquey** [reply]
See https://neper.info/papers.html#id2.

**Reply by Laowulve** [reply]
Appreciate your reply. I'll try it.
